### PR TITLE
[#LR-1679] Set Savon open/read timeouts from config, with defaults

### PR DESCRIPTION
**Pull request by adamdullenty**

### CODE REVIEW

File: lib/dpd_api/config.rb [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
+    attr_writer :soap_endpoint,
+      :service_wsdl,
+      :logger
+
+    attr_accessor :username,
+      :password,
+      :sandbox,
+      :adapter,
+      :open_timeout,
+      :read_timeout
```
Comment (adamdullenty) [inline lib/dpd_api/config.rb:12]: 🆕

File: lib/dpd_api/request_handler.rb [hunk 2]
```diff
@@ -28,8 +28,8 @@ def client
         wsdl: config.service_wsdl[request_name],
         endpoint: config.soap_endpoint[request_name],
         namespace: config.soap_endpoint[request_name],
-        open_timeout: 600,
-        read_timeout: 600,
+        open_timeout: config.open_timeout || 10, # Seconds
+        read_timeout: config.read_timeout || 30, # Seconds
```
Comment (adamdullenty) [inline lib/dpd_api/request_handler.rb:32]: Generally, DPD API responds pretty quickly - usually ~1s or less: [Datadog](https://app.datadoghq.com/apm/traces?query=%40_top_level%3A1%20env%3Aproduction%20%40network.destination.ip%3Apublic-ws.dpd.com&agg_m=%40duration&agg_m_source=base&agg_t=max&analyticsOptions=%5B%22line%22%2C%22dog_classic%22%2Cnull%2Cnull%5D&cols=service%2Cresource_name%2C%40http.url%2C%40duration%2C%40http.method%2C%40http.status_code%2C%40_span.count%2C%40_duration.by_service&fromUser=true&graphType=flamegraph&historicalData=true&messageDisplay=inline&query_translation_version=v0&shouldShowLegend=true&sort=time&spanType=service-entry&traceQuery=&view=spans&viz=timeseries&start=1719398668677&end=1720608268677&paused=false)

it's only in the recent spike that this went up to several minutes

File: dpd.gemspec [hunk 3]
```diff
@@ -23,6 +23,7 @@ Gem::Specification.new do |spec|
 
   spec.add_dependency "savon", "~> 2.13.0"
   spec.add_dependency "httpclient"
+  spec.add_dependency "rack", "2.2.9"
```
Comment (adamdullenty) [inline dpd.gemspec:26]: Pins to the current version used by the monolith

This resolves an incompatibility issue between Savon and the later version of Rack that was causing tests to fail

**Review by alypeng — APPROVED** [review]
🧼

**Review by ivantoniev — APPROVED** [review]

**Pull request merged**